File: tests/fixtures/sample.py
import os
import json  # unused

def used_function():
    return os.getcwd()

def dead_function():
    return "nobody calls me"

class UsedClass:
    def method(self):
        return used_function()

class DeadClass:
    pass

active_var = UsedClass()
dead_var = "unused"

result = active_var.method()

File: tests/fixtures/sample.js
import { readFileSync } from 'fs';
import { join } from 'path'; // unused

function usedFunction() {
  return readFileSync('file.txt', 'utf8');
}

function deadFunction() {
  return "nobody calls me";
}

const activeVar = usedFunction();
const deadVar = "unused";

class UsedClass {
  run() {
    return activeVar;
  }
}

class DeadClass {
  noop() {}
}

const instance = new UsedClass();
instance.run();
